test/infra_test.go

```diff
@@ -1,126 +1,228 @@
 package test
 
 import (
+	"context"
 	"crypto/tls"
+	"encoding/base64"
 	"fmt"
+	"log"
+	"path"
 	"strings"
 	"testing"
 	"time"
 
-	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/eks"
 	"github.com/gruntwork-io/terratest/modules/k8s"
-	"github.com/gruntwork-io/terratest/modules/random"
 	"github.com/gruntwork-io/terratest/modules/terraform"
 	"github.com/stretchr/testify/assert"
+
+	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/homedir"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/token"
 )
 
-var namespace string = "monitoring"
+var namespace string = "awesomeeyes"
 var kubeOptions *k8s.KubectlOptions = k8s.NewKubectlOptions("", "", namespace)
 
 // Setup a TLS configuration to submit with the helper, a blank struct is acceptable
 var tlsConfig tls.Config = tls.Config{}
 
-// Returns pod name
-func awaitPods(t *testing.T, kubeOptions *k8s.KubectlOptions, filter string) string {
-	var podName string
+// https://stackoverflow.com/questions/60547409/unable-to-obtain-kubeconfig-of-an-aws-eks-cluster-in-go-code/60573982#60573982
+func newClientset(cluster *eks.Cluster) (*kubernetes.Clientset, error) {
+	const (
+		RecommendedHomeDir  = ".kube"
+		RecommendedFileName = "config"
+	)
 
-	pods := k8s.ListPods(t, kubeOptions, v1.ListOptions{FieldSelector: "status.phase=Running"})
+	var RecommendedHomeFile = path.Join(homedir.HomeDir(), RecommendedHomeDir, RecommendedFileName)
 
-	for _, pod := range pods {
+	// log.Printf("%+v", cluster)
 
-		// Await all deployed pods to be available and ready
-		k8s.WaitUntilPodAvailable(t, kubeOptions, pod.Name, 60, 1*time.Second)
+	gen, err := token.NewGenerator(true, false)
+	if err != nil {
+		return nil, err
+	}
 
-		if strings.Contains(pod.Name, filter) {
-			podName := pod.Name
-			return podName
-		}
+	opts := &token.GetTokenOptions{
+		ClusterID: aws.StringValue(cluster.Name),
 	}
 
-	return podName
-}
+	tok, err := gen.GetWithOptions(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	ca, err := base64.StdEncoding.DecodeString(aws.StringValue(cluster.CertificateAuthority.Data))
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(
+		&rest.Config{
+			Host:        aws.StringValue(cluster.Endpoint),
+			BearerToken: tok.Token,
+			TLSClientConfig: rest.TLSClientConfig{
```

Comment:
im on mobile so i cant check, but looks like indentation is not right. please make sure that you run `go fmt` on whole file